lang/cs/Org.Apache.REEF.Tang/Protobuf/ProtocolBufferClassHierarchy.cs

```diff
@@ -59,9 +61,12 @@ private static Org.Apache.REEF.Tang.Protobuf.Node SerializeNode(INode n)
         {
             IList<Org.Apache.REEF.Tang.Protobuf.Node> children = new List<Org.Apache.REEF.Tang.Protobuf.Node>();
 
-            foreach (INode child in n.GetChildren())
+            lock (childIteratorLock)
```

Comment:
Wouldn't you also need a lock around the other places where the children could be mutated?